plugin-interface/src/main/scala/mesosphere/marathon/plugin/http/HttpRequestHandlerBase.scala

```diff
@@ -0,0 +1,56 @@
+package mesosphere.marathon.plugin.http
+
+import java.io.{ ByteArrayOutputStream, InputStream, OutputStream }
+import java.net.{ URL, URLConnection }
+import java.nio.file.{ Files, Paths }
+
+import scala.annotation.tailrec
+import scala.util.Try
+
+//scalastyle:off magic.number
+abstract class HttpRequestHandlerBase extends HttpRequestHandler {
+
+  protected[this] def serveResource(path: String, response: HttpResponse): Unit = {
+    def transfer(in: InputStream, out: OutputStream) {
```

Comment:
Forget what I said about transferTo, that's only useful if you're writing directly to a socket (using a Channel).

But Marathon already depends on Guava, so we can get rid of this method and do:

```scala
val content = withResource(path) { url =>
  response.body(mediaMime(url), Resources.toByteArray(url))
  response.status(200)
}
content.getOrElse(response.status(404))
```